src/System.d.ts

```diff
@@ -1,29 +1,70 @@
 import {Component, ComponentConstructor} from "./Component";
+import { Entity } from "./Entity";
 
 /**
  * A system that manipulates entities in the world.
  */
 export abstract class System {
+  /**
+   * Defines what Components the System will query for.
+   * This needs to be user defined.
+   */
+  static queries: {
+    [queryName: string]: {
+      components: (Component | NotComponent)[],
+      listen?: {
+        added?: Boolean,
+        removed?: Boolean,
+        changed?: Boolean | Component[],
+      },
+    }
+  };
```

Comment:
We need to add a `mandatory?: boolean` property. Also according to TS documentation I think we should be using the `boolean` type and not `Boolean`. See https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

I'd be happy to make the fix if the maintainers show signs of wanting to merge this PR in the near future.

Reply:
@snowfrogdev I'd be happy to get this merge if you are willing to update it. I'm not a typescript dev so I couldn't help on the feedback on the current PR but if you feel the current change is useful, as it's really just changing the .ts definitions, I'm happy to get this merge asap and we could fix the rest of the issues in future PRs.

Reply:
@fernandojsg I work almost exclusively with TypeScript and I'll be happy to help out. It's safe for you to merge this PR in at any time. It is a big improvement over what we have at the moment. Like you said, we can make further improvements in future PRs. Thanks for the quick response by the way. I just started using your library and I was a bit worried you had gone dormant. 😉